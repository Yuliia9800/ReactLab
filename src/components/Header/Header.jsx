import React from 'react';

import { Button } from 'common';
import { LOGOUT_BUTTON_TEXT } from 'constants/constants';
import Logo from './components/Logo/Logo';

function Header() {
	return (
		<div className='primary-content navbar px-5'>
			<div className='flex-1'>
				<Logo />
			</div>
			<div className='flex-none gap-6'>
				<p className='font-bold'>Yuliia</p>
				<Button
					buttonText={LOGOUT_BUTTON_TEXT}
					className='btn-ghost'
					onClick={() => console.log('HeaderButtonClick')}
				/>
			</div>
		</div>
	);
}

export default Header;
