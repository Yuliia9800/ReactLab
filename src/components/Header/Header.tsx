import React, { useEffect, useState } from 'react';
import { Outlet, useLocation, useNavigate } from 'react-router-dom';

import { Button } from 'common';
import { LOGOUT_BUTTON_TEXT } from 'constant';
import Logo from './components/Logo/Logo';

function Header() {
	const navigate = useNavigate();
	const location = useLocation();
	const [token, setToken] = useState(null);

	useEffect(() => {
		const token = localStorage.getItem('token');

		setToken(token);
	}, [location]);

	const handleButtonClick = () => {
		localStorage.removeItem('token');
		navigate('/login');
	};

	return (
		<>
			<div className='primary-content navbar px-5'>
				<div className='flex-1'>
					<Logo />
				</div>
				<div className='flex-none gap-6'>
					{token ? (
						<>
							<p className='font-bold'>Yuliia</p>
							<Button
								buttonText={LOGOUT_BUTTON_TEXT}
								className='btn-ghost'
								onClick={handleButtonClick}
							/>
						</>
					) : null}
				</div>
			</div>
			<Outlet />
		</>
	);
}

export default Header;
