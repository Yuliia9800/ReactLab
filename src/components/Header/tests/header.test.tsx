import React from 'react';

import { renderWithProviders, screen } from 'testing';
import Header from '../Header';

const state = {
	user: {
		name: 'Yuliia',
		token: 'token',
	},
};

test('should match snapshot', () => {
	const { container } = renderWithProviders(<Header />, {
		state,
	});

	expect(container).toMatchSnapshot();
});

test("should have logo and user's name", () => {
	renderWithProviders(<Header />, {
		state,
	});

	expect(screen.getByText('Yuliia')).toBeInTheDocument();
	expect(screen.getByRole('img', { name: /logo/i })).toBeInTheDocument();
});
