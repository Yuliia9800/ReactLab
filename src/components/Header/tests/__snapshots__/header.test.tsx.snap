// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="primary-content navbar px-5"
  >
    <div
      class="flex-1"
    >
      <a
        href="/courses"
      >
        <img
          alt="logo"
          class="w-40"
          src="logo.webp"
        />
      </a>
    </div>
    <div
      class="flex-none gap-6"
    >
      <p
        class="font-bold"
      >
        Yuliia
      </p>
      <button
        class="btn-sm btn px-5 btn-ghost"
        type="button"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
